### PR TITLE
container fixes

### PR DESCRIPTION
**Pull request by Maoni0**
fix for running in a container without mem limit set on Windows;
also fixing the LocalGC standalone case on Linux

@brianrob @andy-ms

### CODE REVIEW

File: src/gc/unix/cgroup.cpp [hunk 1]
```diff
@@ -437,7 +437,18 @@ size_t GetRestrictedPhysicalMemoryLimit()
     size_t physical_memory_limit;
  
     if (!CGroup::GetPhysicalMemoryLimit(&physical_memory_limit))
```
Comment (andy-ms) [inline src/gc/unix/cgroup.cpp]: This might be neater using `||` instead of nested `if`s

Comment (jkotas) [reply src/gc/unix/cgroup.cpp]: Or drop the `else`, like:

```
if (!CGroup::GetPhysicalMemoryLimit(&physical_memory_limit))
    return 0;

// If there's no memory limit specified on the container this 
// actually returns 0x7FFFFFFFFFFFF000 (2^63-1 rounded down to 
// 4k which is a common page size). So we know we are not
// running in a memory restricted environment.
if (physical_memory_limit > 0x7FFFFFFF00000000)
    return 0;
```

Comment (Maoni0) [reply src/gc/unix/cgroup.cpp]: done!

**Comment by Maoni0** [comment]
@dotnet-bot test coreclr-ci (Test Pri0 Linux x64 checked)

**Comment by Maoni0** [comment]
@dotnet-bot test CentOS7.1 x64 Checked Innerloop Build and Test 
@dotnet-bot test CentOS7.1 x64 Debug Innerloop Build

**Comment by Maoni0** [comment]
@dotnet-bot test CentOS7.1 x64 Checked Innerloop Build and Test

**Pull request merged**